flambe/field/text.py

```diff
@@ -138,62 +199,100 @@ def vocab_size(self) -> int:
-        data : Iterable[str]
-            List of input strings.
+        *data: np.ndarray
+            The data
 
         """
-        # Iterate over all examples
         examples: Iterable = (e for dataset in data for e in dataset if dataset is not None)
-        embeddings_matrix: List[torch.Tensor] = []
-        model = self.model
-
-        if model is not None:
-            if self.setup_all_embeddings:
-                examples = chain(examples, model.vocab.keys())
-
-            # Add embeddings for special tokens
-            for special in self.specials:
-                if special in model:
-                    embeddings_matrix.append(torch.tensor(model[special]))
-                else:
-                    embeddings_matrix.append(torch.randn(model.vector_size))
 
-        # Get current last id
         index = len(self.vocab) - 1
 
+        # First load special tokens
+        for token in self.specials:
+            if token not in self.vocab:
+                self.vocab[token] = index = index + 1
+
         for example in examples:
             # Lowercase if requested
             example = example.lower() if self.lower else example
             # Tokenize and add to vocabulary
             for token in self.tokenizer(example):
                 if token not in self.vocab:
-                    if model is not None:
-                        if token in model:
-                            self.vocab[token] = index = index + 1
-                            embeddings_matrix.append(torch.tensor(model[token]))
-                        else:
-                            if self.unk_init_all:
-                                # Give every OOV it's own embedding
-                                self.vocab[token] = index = index + 1
-                                embeddings_matrix.append(torch.randn(model.vector_size))
-                            else:
-                                # Collapse all OOV's to the same token
-                                # id
-                                self.vocab[token] = self.vocab[self.unk]
-                            self.unk_numericals.add(self.vocab[token])
+                    self.vocab[token] = index = index + 1
+
+    def _build_embeddings(self, model: KeyedVectors) -> Tuple[odict, torch.Tensor]:
+        """
+        Create the embeddings matrix and the new vocabulary in
+        case this objects needs to use an embedding model.
+
+        A new vocabulary needs to be built because of the parameters
+        that could allow, for example, collapsing OOVs.
+
+        Parameters
+        ----------
+        model: KeyedVectors
+            The embeddings
+
+        Returns
+        -------
+        Tuple[OrderedDict, torch.Tensor]
+            A tuple with the new embeddings and the embedding matrix
+        """
+        embedding_matrix: List[torch.Tensor] = []
+        new_vocab: odict[str, int] = odict()
+
+        new_index = -1
+
+        for token, index in self.vocab.items():
```

Comment:
Can this be a single loop since the `index` value is unused?

Suggestion:
```
tokens = self.vocab.values()
if self.setup_all_embeddings:
    tokens.update(model.vocab.keys())

for token in tokens:
    ...

Reply:
Sorry missed this!

Reply:
Yes, excellent.